tools/system_libs.py

```diff
@@ -413,7 +414,7 @@ def get_ext(self):
     """
     Return the appropriate file extension for this library.
     """
-    return '.a' if shared.Settings.WASM_BACKEND and shared.Settings.WASM_OBJECT_FILES else '.bc'
+    return '.a' if shared.Settings.WASM_BACKEND else '.bc'
```

Comment:
Does this mean we'd get a `.a` file with bc files inside it for the wasm backend? If so, is NoBCLibrary even needed?

Reply:
Yes.  It means with the wasm backend we only ever get `.a` files, which is really how it should have always been.

`NoBCLibrary` exists solely for fastcomp.  I disabled fastcomp's default behaviour of not using `ar` archives but rather linking to large `.bc` objects.